src/views/Dashboard/Customer/Home/MyTest/TestResult.vue

```diff
@@ -81,45 +88,232 @@
+        }
+      } catch (error) {
+        this.resultLoading = false;
+      }
+    },
+
+    async decryptWallet() {
+      try {
+        this.publicKey = this.lab.info.box_public_key;
+        if (this.actionType === "result") {
+          await this.parseResult();
+        }
+
+        if (this.actionType === "download") {
+          await this.downloadDecryptedFromIPFS();
+        }
+      } catch (error) {
+        console.log(error);
+      }
+    },
+
+    async parseResult() {
+      try {
+        const path = this.files[0].fileLink.replace(this.baseUrl, "");
+        const secretKey = this.privateKey;
+        const publicKey = this.lab.info.box_public_key;
+        
+        const pair = {
+          secretKey,
+          publicKey
+        };
+
+        const typeFile = "text/plain";
+        const channel = new MessageChannel();
+        channel.port1.onmessage = ipfsWorker.workerDownload;
+        ipfsWorker.workerDownload.postMessage({ path, pair, typeFile }, [
+          channel.port2
+        ]);
+
+        ipfsWorker.workerDownload.onmessage = (event) => {
+          const regexMatchPdf = /^(data:application\/\pdf)/g 
+          const isDataPdf = regexMatchPdf.test(event.data);
+          this.isDataPdf = isDataPdf;
+          
+          this.result = event.data;
+          this.resultLoading = false;
+        };
+      } catch (error) {
+        this.resultLoading = false;
+      }
+    },
+
+    async downloadDecryptedFromIPFS() {
+      try {
+        const channel = new MessageChannel();
+        channel.port1.onmessage = ipfsWorker.workerDownload;
+        const path = this.files[this.fileDownloadIndex].fileLink.replace(
+          this.baseUrl,
+          ""
+        );
+
+        const secretKey = this.privateKey;
+        const publicKey = this.publicKey;
+        const pair = {
+          secretKey,
+          publicKey
+        };
+        const typeFile = "text/plain";
+      
+        ipfsWorker.workerDownload.postMessage({ path, pair, typeFile }, [
+          channel.port2
+        ]);
+
+        ipfsWorker.workerDownload.onmessage = (event) => {
+          this.download(
+            event.data,
+            this.files[this.fileDownloadIndex].fileName
+          );
+        };
+      } catch (error) {
+        console.log(error);
+      }
+    },
+
+    async download(data, fileName) {
+      const blob = new Blob([data], { type: "text/plain" });
+      const e = document.createEvent("MouseEvents");
+      const a = document.createElement("a");
+      a.download = fileName;
+      a.href = this.isDataPdf ? data : window.URL.createObjectURL(blob);
+      a.dataset.downloadurl = ["text/json", a.download, a.href].join(":");
+      e.initMouseEvent(
+        "click",
+        true,
+        false,
+        window,
+        0,
+        0,
+        0,
+        0,
+        0,
+        false,
+        false,
+        false,
+        false,
+        0,
+        null
+      );
+      a.dispatchEvent(e);
+    },
```

Comment:
I think these are duplicate functions of the ones you can find here: https://github.com/debionetwork/debio-customer-ui/blob/d239502d0083230a8488a84e1396198f8a4914b2/src/common/lib/ipfs/index.js#L41

Please reconsider and try to use the methods in the files mentioned instead.

Reply:
Oke, I will fix it ASAP.